cmd/certs.go

```diff
@@ -90,19 +94,70 @@ func getRootCAs(certsCAsDir string) (*x509.CertPool, error) {
 	return rootCAs, nil
 }
 
-func getSSLConfig() (publicCerts []*x509.Certificate, rootCAs *x509.CertPool, secureConn bool, err error) {
-	if !(isFile(getPublicCertFile()) && isFile(getPrivateKeyFile())) {
-		return publicCerts, rootCAs, secureConn, err
+func parsePKCS12CertFile(pkcs12File string) (x509Certs []*x509.Certificate, tlsCert *tls.Certificate, err error) {
+	// Read pkcs12 file.
+	var data []byte
+	if data, err = ioutil.ReadFile(pkcs12File); err != nil {
+		return nil, nil, err
+	}
+
+	// Decode if the content is base64 encoded.
+	var decodedData []byte
+	if decodedData, err = ioutil.ReadAll(base64.NewDecoder(base64.StdEncoding, bytes.NewBuffer(data))); err == nil {
```

Comment:
It is better to fail here if err != nil  - rather than getting overwritten by the next `pkcs12.ToPEM()`